### PR TITLE
chore(e2e): use rustls instead of openssl

### PR DESCRIPTION
**Pull request by FlorentinDUBOIS**

### CODE REVIEW

File: e2e/src/mock/https_client.rs [hunk 1]
```diff
@@ -1,20 +1,48 @@
...
     StatusCode,
 };
-use hyper_tls::HttpsConnector;
+use hyper_rustls::HttpsConnector;
+use rustls::{
+    client::{ClientConfig, ServerCertVerified, ServerCertVerifier},
+    Certificate, ServerName,
+};
+
+// We implements our own verifier to allow self-signed certificates
```
Comment (Keksoj) [inline e2e/src/mock/https_client.rs:14]: remove the s typo

**Review by Keksoj — CHANGES_REQUESTED** [review]

**Review by Wonshtrum — APPROVED** [review]
Looks good to me!

**Comment by Keksoj** [comment]
Look great to me overall, typo aside

File: e2e/src/mock/https_client.rs (hunk 1 above)
Comment (FlorentinDUBOIS) [reply e2e/src/mock/https_client.rs:14]: Thx!

**Review by Keksoj — APPROVED** [review]

**Pull request merged**